arithmetization/src/main/java/net/consensys/linea/zktracer/ZkTracer.java

```diff
@@ -54,11 +63,53 @@ public ZkTracer() {
     this.hub = new Hub();
   }
 
-  public ZkTrace getTrace() {
-    for (Module module : this.hub.getModulesToTrace()) {
-      zkTraceBuilder.addTrace(module);
+  public Path writeToTmpFile() {
+    try {
+      final Path traceFile = Files.createTempFile(null, ".lt");
+      this.writeToFile(traceFile);
+      return traceFile;
+    } catch (IOException e) {
+      throw new RuntimeException(e);
     }
-    return zkTraceBuilder.build();
+  }
+
+  @Override
+  public void writeToFile(final Path filename) {
+    log.warn("[TRACING] Starting serialization to " + filename.toAbsolutePath());
+    Stopwatch sw = Stopwatch.createStarted();
+
+    final List<Module> modules = this.hub.getModulesToTrace();
+    final List<ColumnHeader> traceMap =
+        modules.stream().flatMap(m -> m.columnsHeaders().stream()).toList();
+    final int headerSize = traceMap.stream().mapToInt(ColumnHeader::headerSize).sum() + 4;
```

Comment:
What's that `+4` ?

Reply:
An i32 for the number of columns to read from the trace

```
header.putInt(traceMap.size());
```